pkg/state/impl/etcd/etcd.go

```diff
@@ -587,12 +573,16 @@ func (st *State) watchKind(ctx context.Context, resourceKind resource.Kind, sing
 	watchCh := st.cli.Watch(ctx, etcdKey, clientv3.WithPrefix(), clientv3.WithPrevKV(), clientv3.WithRev(revision))
 
 	go func() {
+		clearbootstrapList := func() { bootstrapList = nil }
+
 		defer func() {
+			clearbootstrapList()
+			cancel()
```

Comment:
this is clearly wrong, `cancel` should happen before draining the channel

Reply:
ok, ignore that, it happens before, but it's confusing in a diff

Reply:
Yeah, so there is actually no reason for two defers here, since those are not conditionals and can be squashed together.